docs/benchmarking-workflow-corefx.md

```diff
@@ -0,0 +1,253 @@
+
+```cmd
+C:\Projects\corefx> build -Release /p:CoreCLROverridePath="C:\Projects\coreclr\bin\Product\Windows_NT.x64.Release"
+```
+
+After the update, the reported CoreCLR version should be changed:
+
+```log
+  [Host]     : .NET Core 3.0.0-preview-27122-01 (CoreCLR 4.6.27121.03, CoreFX 4.7.18.57103), 64bit RyuJIT
+  Job-CYVJFZ : .NET Core ? (CoreCLR 4.6.27517.0, CoreFX 4.7.19.11901), 64bit RyuJIT
+```
+
+**It's very important to validate the reported version numbers** to make sure that you are running the benchmarks using the versions you think that you are using.
+
+## Benchmarking new API
+
+When developing new CoreFX features, we should be thinking about the performance from day one. One part of doing this is writing benchmarks at the same time when we write our first unit tests. Keeping the benchmarks in a separate repository makes it a little bit harder to run the benchmarks against new CoreFX API, but it's still very easy.
+
+### Reference
+
+When you develop a new feature, whether it's a new method/type/library in CoreFX all you need to do is to build it in Release and just reference the produced `.dll` from the [MicroBenchmarks.csproj](../src/benchmarks/micro/MicroBenchmarks.csproj) project file.
+
+The easiest way to do it is to open [MicroBenchmarks.sln](../src/benchmarks/micro/MicroBenchmarks.sln) with Visual Studio, right click on the [MicroBenchmarks](../src/benchmarks/micro/MicroBenchmarks.csproj) project file, select "Add", then "Reference..." and in the new Dialog Window click "Browse" in the left bottom corner. From the File Picker, choose the new library and click "Add". Please don't forget to Save the changes (Ctrl+Shift+S). From this moment you should be able to consume new public types and methods exposed by the referenced library.
+
+Sample changes:
+
+```cs
+namespace System
+{
+    public static class Console
+    {
+        public static void WriteHelloWorld() => WriteLine("Hello World!");
+        
+        // the rest omitted for brevity
+    }
+}
+```
+
+Sample project file change:
+
+```xml
+<ItemGroup>
+  <Reference Include="System.NewAPI">
+    <HintPath>..\..\..\..\corefx\artifacts\bin\runtime\netcoreapp-Windows_NT-Release-x64\System.Console.dll</HintPath>
+  </Reference>
+</ItemGroup>
+```
+
+### PR
+
+Keeping the benchmarks in a separate repository requires us to send a separate PR to the performance repo.
+
+The first thing you need to do is to send a PR with the new API to the CoreFX repository. Once your PR gets merged and a new NuGet package is published to CoreFX NuGet feed you should remove the Reference to a `.dll` and install/update the package consumed by [MicroBenchmarks](../src/benchmarks/micro/MicroBenchmarks.csproj). After that, you should send a PR to the performance repo.
+
+We are aware of the fact that it's not a perfect workflow, however, we think that such a situation is going to be rare enough to not be a problem.
```

Comment:
Why would this be rare? Don't we want a growing body of relevant performance tests for new APIs/features?

Reply:
Good point, I have removed this sentence.